### DISCUSSION TITLE
Parameterisation of activation energy

### DISCUSSION THREAD

**Discussion by kyang0607** [pybamm-team/PyBaMM#2581] [General]
Dear all, 

I have one question about the parameter modification of activation energy. Some paramters are dependent on temperature and defined as functions.  Like this one: 
def nmc_LGM50_electrolyte_exchange_current_density_ORegan2022(
    c_e, c_s_surf, c_s_max, T
):
    """
    Exchange-current density for Butler-Volmer reactions between NMC and LiPF6 in
    EC:DMC.

    References
    ----------
    .. [1] Kieran O’Regan, Ferran Brosa Planella, W. Dhammika Widanage, and Emma
    Kendrick. "Thermal-electrochemical parameters of a high energy lithium-ion
    cylindrical battery." Electrochimica Acta 425 (2022): 140700

    Parameters
    ----------
    c_e : :class:`pybamm.Symbol`
        Electrolyte concentration [mol.m-3]
    c_s_surf : :class:`pybamm.Symbol`
        Particle concentration [mol.m-3]
    c_s_max : :class:`pybamm.Symbol`
        Maximum particle concentration [mol.m-3]
    T : :class:`pybamm.Symbol`
        Temperature [K]

    Returns
    -------
    :class:`pybamm.Symbol`
        Exchange-current density [A.m-2]
    """
    i_ref = 5.028  # (A/m2)
    alpha = 0.43
    E_r = 2.401e4
    arrhenius = exp(E_r / constants.R * (1 / 298.15 - 1 / T))

    c_e_ref = Parameter("Typical electrolyte concentration [mol.m-3]")

    return (
        i_ref
        * arrhenius
        * (c_e / c_e_ref) ** (1 - alpha)
        * (c_s_surf / c_s_max) ** alpha
        * (1 - c_s_surf / c_s_max) ** (1 - alpha)
    )
In this function, parameters like i_ref/alpha/E_r are hard-coded. I would like to test the range of these paramters. Is there a easy way to modify the parameters?

**Comment by brosaplanella** [comment]
You can write your custom exchange current density to adjust the value you want (see [this notebok](https://github.com/pybamm-team/PyBaMM/blob/develop/examples/notebooks/parameterization/parameterization.ipynb) for more details). If you need to change the activation energy often, you can define it as a `Parameter` in the function and then add that parameter to the dictionary (like the other parameters).

**Reply by kyang0607** [reply]
Thanks a lot. I will try this later.
